### PR TITLE
feat: migrate download toggle component

### PR DESCRIPTION
**Pull request by araujogui**
## Description

Migrate `DownloadToggle` component

## Related Issues

Related to #5303 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/Downloads/DownloadToggle/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import classnames from 'classnames';
+import styles from './index.module.scss';
+
+export type Props = {
+  handleClick: (type: string) => void;
+  selected: string;
+  showDescription?: boolean;
+};
+
+const DownloadToggle = ({
```
Comment (AugustinMauroy) [inline components/Downloads/DownloadToggle/index.tsx]: You should use FC ref: "contribution guideline"

File: i18n/locales/en.json [hunk 2]
```diff
@@ -67,5 +67,8 @@
   "components.sections.newHeader.links.apiDocs": "API Docs",
   "components.sections.newHeader.links.about": "About",
   "components.sections.newHeader.links.download": "Downloads",
-  "components.sections.newHeader.links.certification": "Certification"
+  "components.sections.newHeader.links.certification": "Certification",
+  "components.downloadToggle.current": "Current",
```
Comment (shanpriyan) [inline i18n/locales/en.json]: keys should be as per  [translation guidelines](https://github.com/nodejs/nodejs.org/blob/572c1edd3eaaf32c6f803894ad54b9e4b219110d/TRANSLATION.md?plain=1#L48)
```suggestion
  "components.downloads.downloadToggle.current": "Current",
```

**Review by shanpriyan — DISMISSED** [review]

File: .eslintjscache [hunk 3]
```diff
@@ -0,0 +1 @@
...sc":"757"},{"messageId":"758","fix":"779","desc":"760"},"suggestUnknown",{"range":"780","text":"781"},"Use `unknown` instead, this will force you to explicitly, and safely assert the type is correct.","suggestNever",{"range":"780","text":"782"},"Use `never` instead, this is useful when instantiating generic type parameters that you don't need to know the type of.","Update the dependencies array to be: [codeArray, langIndex, language]",{"range":"783","text":"784"},"directive","",{"range":"785","text":"781"},{"range":"785","text":"782"},"suggestOptionalChain",{"range":"786","text":"787"},"Consider using the optional chain operator `?.` instead. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator.",{"range":"788","text":"781"},{"range":"788","text":"782"},{"range":"789","text":"781"},{"range":"789","text":"782"},{"range":"790","text":"781"},{"range":"790","text":"782"},{"range":"791","text":"781"},{"range":"791","text":"782"},{"range":"792","text":"781"},{"range":"792","text":"782"},[1025,1028],"unknown","never",[1745,1756],"[codeArray, langIndex, language]",[372,375],[389,390],"?",[248,251],[491,494],[690,693],[1098,1101],[683,686]]
```
Comment (mikeesto) [inline .eslintjscache]: Can you please remove this file from the PR?

File: .eslintmdcache [hunk 4]
```diff
@@ -0,0 +1 @@
...org/pages/en/blog/npm/peer-dependencies.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/en/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/es/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/fr/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/id/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/ja/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/ka/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/ko/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/nl/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/pt-br/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/ro/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/ru/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/tr/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/uk/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/zh-cn/docs/index.mdx",[],[],"/Users/araujo/Documents/Projects/nodejs.org/pages/zh-tw/docs/index.mdx",[],[]]
```
Comment (mikeesto) [inline .eslintmdcache]: Same with this one

**Review by mikeesto — CHANGES_REQUESTED** [review]

**Comment by ovflowd** [comment]
@araujogui could you check those files that @mikeesto mentioned? 👀

**Review by mikeesto — APPROVED** [review]
LGTM, thanks

**Review by aymen94 — APPROVED** [review]

**Pull request merged**